server/src/main/java/com/cloud/storage/StoragePoolAutomationImpl.java

```diff
@@ -117,9 +117,11 @@ public boolean maintain(DataStore store) {
                 spes = primaryDataStoreDao.listBy(pool.getDataCenterId(), pool.getPodId(), pool.getClusterId(), ScopeType.CLUSTER);
             }
             for (StoragePoolVO sp : spes) {
-                if (sp.getStatus() == StoragePoolStatus.PrepareForMaintenance) {
-                    throw new CloudRuntimeException("Only one storage pool in a cluster can be in PrepareForMaintenance mode, " + sp.getId() +
-                        " is already in  PrepareForMaintenance mode ");
+                if (sp.getParent() != pool.getParent()) { // If Datastore cluster is tried to prepare for maintenance then child storage pools are also kept in PrepareForMaintenance mode
+                    if (sp.getStatus() == StoragePoolStatus.PrepareForMaintenance) {
+                        throw new CloudRuntimeException("Only one storage pool in a cluster can be in PrepareForMaintenance mode, " + sp.getId() +
+                                " is already in  PrepareForMaintenance mode ");
+                    }
```

Comment:
good comment but it should be reflected in a method name, modularising the code further.